src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  injure(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health <= 0);
```

Comment:
[CODE PERFORMANCE] The filter function is incorrectly used here. It should be used to remove the dead animal from the array, not to keep it. The condition should check for animals that are still alive (health > 0) instead of dead (health <= 0).